weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/AddToDatasetDrawer.tsx

```diff
@@ -0,0 +1,309 @@
+            <Typography
+              sx={{
+                ...typographyStyle,
+                fontWeight: 600,
+                fontSize: '1.25rem',
+              }}>
+              Add example{selectedCalls.length !== 1 ? 's' : ''} to dataset
+            </Typography>
+          </Box>
+          <Box sx={{display: 'flex', gap: 1}}>
+            {currentStep === 2 && (
+              <Button
+                onClick={() => setIsFullscreen(!isFullscreen)}
+                variant="ghost"
+                icon="full-screen-mode-expand"
+                tooltip={isFullscreen ? 'Exit fullscreen' : 'Fullscreen'}
+                size="medium"
+              />
+            )}
+            {currentStep === 1 && (
+              <Button
+                size="medium"
+                variant="ghost"
+                icon="close"
+                onClick={onClose}
+                tooltip="Close"
+              />
+            )}
+          </Box>
+        </Box>
+        <Typography
+          sx={{
+            ...typographyStyle,
+            color: 'text.secondary',
+            px: 2,
+            pt: 2,
+            pb: 0,
+          }}>
+          Step {currentStep} of 2:{' '}
+          {currentStep === 1 ? 'Select dataset & map fields' : 'Edit & confirm'}
+        </Typography>
+        <Box
+          sx={{
+            flex: 1,
+            overflow: 'auto',
+            px: currentStep === 1 ? 2 : 0,
+            display: 'flex',
+            flexDirection: 'column',
+          }}>
+          {error && (
+            <Box sx={{mb: 2, p: 2, bgcolor: 'error.light', borderRadius: 1}}>
+              <Typography sx={typographyStyle} color="error.dark">
+                {error}
+              </Typography>
+            </Box>
+          )}
+          {renderStepContent()}
+        </Box>
+        <Box
+          sx={{
+            p: 2,
+            borderTop: '1px solid',
+            borderColor: 'divider',
+            display: 'flex',
+            justifyContent: 'space-between',
+            gap: 1,
+            bgcolor: 'background.paper',
+          }}>
+          {currentStep === 1 ? (
+            <Button
+              onClick={handleNext}
+              color="primary"
+              variant="primary"
+              disabled={isNextDisabled}
+              style={{width: '100%'}}
+              twWrapperStyles={{width: '100%'}}>
+              Next
+            </Button>
+          ) : (
+            <>
+              <Button
+                onClick={handleBack}
+                color="primary"
+                variant="secondary"
+                style={{width: '100%'}}
+                twWrapperStyles={{width: '100%'}}>
+                Back
+              </Button>
+              <Button
+                onClick={handleCreate}
+                color="primary"
+                variant="primary"
+                style={{width: '100%'}}
+                twWrapperStyles={{width: '100%'}}>
+                Add to dataset
```

Comment:
Please add a loading indicator / progress after clicking add to dataset. It feels like it is just frozen while the network request completes

Reply:
When the query is running the drawer will now show the weave loader